UseCases/Sources/InstrumentsAppTracing/OSSignpostTracing.swift

```diff
@@ -0,0 +1,267 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Instrumentation
+import Foundation // string conversion for os_log seems to live here
+
+#if os(macOS) || os(tvOS) || os(iOS) || os(watchOS)
+import os.log
+import os.signpost
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: OSSignpost Tracing
+
+@available(OSX 10.14, *)
+@available(iOS 10.0, *)
+@available(tvOS 10.0, *)
+@available(watchOS 3.0, *)
+public struct OSSignpostTracingInstrument: TracingInstrument {
+    let log: OSLog
+    let signpostName: StaticString
+
+    public init(subsystem: String, category: String, signpostName: StaticString) {
+        self.log = OSLog(subsystem: subsystem, category: category)
+        self.signpostName = signpostName
+    }
+
+    // ==== ------------------------------------------------------------------------------------------------------------
+    // MARK: Instrument API
+
+    public func extract<Carrier, Extractor>(_ carrier: Carrier, into baggage: inout BaggageContext, using extractor: Extractor) {
+        // noop; we could handle extracting our keys here
+    }
+
+    public func inject<Carrier, Injector>(_ baggage: BaggageContext, into carrier: inout Carrier, using injector: Injector) {
+        // noop; we could handle injecting our keys here
+    }
+
+    // ==== ------------------------------------------------------------------------------------------------------------
+    // MARK: Tracing Instrument API
+
+    public func startSpan(
+        named operationName: String,
+        context: BaggageContext,
+        ofKind kind: SpanKind,
+        at timestamp: Timestamp?
+    ) -> Span {
+        OSSignpostSpan(
+            log: self.log,
+            named: operationName,
+            signpostName: self.signpostName,
+            context: context
+            // , kind ignored
+            // , timestamp ignored, we capture it automatically
+        )
+    }
+}
```

Comment:
note that some tracer may choose not to expose setting the timestamp explicitly by user.

this is the case with XRay tracer as well, at the moment (I dont really see a use case for that);
arguably its easier to change XRay tracer than `OSSignpostSpan` ;-)

BTW this is not required by OT